flax/errors.py

```diff
@@ -69,67 +69,82 @@ def __init__(self, message):
     error_msg = f'{message} ({error_page}#{module_name}.{class_name})'
     super().__init__(error_msg)
 
+
 #################################################
 # scope.py errors                               #
 #################################################
 
-class InitScopeInvalidRngsError(FlaxError):
+
+class MissingRngError(FlaxError):
+  """
+  This error is raised if a PRNG is required but it is not provided as an input.
+  Please see :class:`InvalidRngError` for more details.
   """
-  When initializing a Module with
-  :meth:`Module.init() <flax.linen.Module.init>`, the first argument can be of
-  two forms:
+  def __init__(self, scope_name, rng_name):
+    super().__init__(f'{scope_name} needs PRNG for "{rng_name}"')
 
-  1. A single PRNGKey. This is in case only one PRNGKey is needed to initialize
-     the ``params`` collection. Note that this::
 
-       SomeModule(...).init(jax.random.PRNGKey(0), ...)
+class InvalidRngError(FlaxError):
```

Comment:
I don't think this error is ever actually raised? Can/should we combine all of the different RNG errors into this one?

Reply:
Good point. I'm changing it so all RNG errors point to here.